docs/developers/NON_S3_COMPATIBLE_BACKENDS.adoc

```diff
@@ -0,0 +1,469 @@
+= Adding support for data backends not supporting the S3 API
+
+These backends are what makes Zenko so valuable: abstracting the complexity of
+multiple APIs to let users work on a single common namespace across multiple
+clouds.
+
+This documents aims at introducing you to the right files in Cloudserver (the
+Zenko stack's subcomponent in charge of API translation, among other things) to
+add support to your own backend of choice.
+
+As usual, should you have any question, please reach out on the
+https://forum.zenko.io[Zenko forum].
+
+== General configuration
+
+There are a number of constants and environment variables to define to support a
+new data backends; here is a list and where to find them:
+
+=== `/constants.js`
+
+- give your backend type a name, as part of the `externalBackends` object;
+- specify whether versioning is implemented, as part of the
+  `versioningNotImplemented` object;
+
+=== `/lib/Config.js`
+
+- this is where you should put common utility functions, like the ones to parse
+  the location object from `locationConfig.json`;
+- make sure you define environment variables (like `GCP_SERVICE_EMAIL` as we'll
+  use those internally for the CI to test against the real remote backend;
+
+=== `/lib/data/external/{{backendName}}Client.js`
+
+- this file is where you'll instantiate your backend client; this should be a
+  class with a constructor taking the config object built in `/lib/Config.js` as
+  parameter;
+- over time, you may need some utility functions which we've defined in the
+  folder `/api/apiUtils`, and in the file `/lib/data/external/utils`;
+
+=== `/lib/data/external/utils.js`
+
+- make sure to add options for `sourceLocationConstraintType` to be equal to
+  the name you gave your backend in `/constants.js`;
+
+=== `/lib/data/external/{{BackendName}}_lib/`
+
+- this folder is where you'll put the functions needed for supporting your
+  backend; keep your files as atomic as possible;
+
+=== [[location-config-test-json]]
+=== `/tests/locationConfig/locationConfigTests.json`
+
+- this file is where you'll create location profiles to be used by your
+  functional tests;
+
+=== `/lib/data/locationConstraintParser.js`
+
+- this is where you'll instantiate your client if the operation the end user
```

Comment:
This sentence is phrased awkwardly, and I'm having trouble understanding the original intent.

Reply:
Trying to say that this is where the new data backend client will be instantiated